common/cache_policy/lru_policy.go

```diff
@@ -50,11 +43,26 @@ func (cache *LRUCache) Get(bk int64) (*CacheBlock, bool) {
 	return cb, found
 }
 
-//Put: Inserts the key,value pair in LRUCache.
-func (cache *LRUCache) Put(key int64, value *CacheBlock) {
+//Resize: resizes a cached block and adjusts occupied size
+func (cache *LRUCache) Resize(bk, newEndIndex int64) bool {
+	var cb *common.Block
+	if node, ok := cache.Elements[bk]; ok {
```

Comment:
is this only for the last block in a file?

Reply:
currently im using it for that, but this can be a resize for any block in the cache.

Reply:
it has to be the end though right? otherwise a block can overlap

Reply:
yes, this policy is separate from streaming - streaming maintains its own state of how the blocks should be placed. This simply caches/resizes whatever blocks are passed down.